parch-key.js

```diff
@@ -0,0 +1,42 @@
+
+class ParchKey {
+    name = 'unassigned';
+    id = 'unassigned';
+
+    /**
+     * @param name A string representation of the name of the top level key object.
+     * @param id A string representation of the id of the top level key object.
+     */
+    constructor(name, id) {
+        // This class is abstract so make sure we cannot instantiate it
+        if (this.constructor === ParchKey) {
+            throw new TypeError('Abstract class "ParchKey" cannot be instantiated directly.');
+        }
+
+        // Make sure the validate function has been defined in subclasses
+        if (undefined === this.validate) {
+            throw new TypeError('Extending class must implement the "validate" function.');
+        }
```

Comment:
I'm not 100% sure about the extension rules for javascript, but since you a validate function below for this class will this ever be used?

Reply:
I removed this if statement.